pymeasure/instruments/anritsu/anritsuMS464xB.py

```diff
@@ -0,0 +1,310 @@
+        validator=strict_range,
+        values=[1, 8000000000],
+    )
+
+    start_frequency = Instrument.control(
+        ":FREQ:STAR?",
+        ":FREQ:STAR %d",
+        """Control starting frequency in Hz.""",
+        validator=strict_range,
+        values=[1, 8000000000],
+    )
+
+    stop_frequency = Instrument.control(
+        ":FREQ:STOP?",
+        ":FREQ:STOP %d",
+        """Control stop frequency in Hz.""",
+        validator=strict_range,
+        values=[1, 8000000000],
+    )
+
+    sweep_type = Instrument.control(
+        ":SWE:TYP?",
+        ":SWE:TYP %s",
+        """Control sweep Type. Valid options are LIN|LOG|FSEGM|ISEGM|POW|MFGC.""",
+        validator=strict_discrete_set,
+        values=_SWEEP_TYPES,
+    )
+
+    sweep_mode = Instrument.control(
+        ":SA:MODE?",
+        ":SA:MODE %s",
+        """ Sweep mode. Valid options are VNA|CLAS.""",
+        validator=strict_discrete_set,
+        values=["VNA", "CLAS"],
+    )
+
+    sweep_points = Instrument.control(
+        ":SWE:POI?",
+        ":SWE:POI %d",
+        """Controls the number of measurement points.""",
+        validator=strict_range,
+        values=[2, 100000],
+    )
+
+    sweep_time = Instrument.control(
+        ":SWE:TIME?",
+        ":SWE:TIME %d",
+        """Controls the sweep time of the indicated channel.""",
+        validator=strict_range,
+        values=[2, 100000],
+    )
+
+    num_averages = Instrument.control(
+        ":AVER:COUN?",
+        ":AVER:COUN %d",
+        """Controls the number of averages.""",
+        validator=strict_range,
+        values=[1, 20000],
+    )
+
+    bandwidth = Instrument.control(
+        ":BWID?",
+        ":BWID %d",
+        """Controls the IF bandwidth.""",
+        validator=strict_range,
+        values=[1, 1000000],
+    )
+
+    def insert_id(self, command):
+        return f":SENS{self.id}{command}"
+
+    def clear_averaging(self):
+        """Clears and restarts the averaging sweep count"""
+        self.write(":AVER:CLE")
+
+
+class AnritsuMS464xB(Instrument):
+    """Represents the Anritsu MS474xB VNA. Implements controls to change the analysis
+    range and to retreve the data for the trace.
+    """
+
+    channels = Instrument.ChannelCreator(
+        VNAChannel, (1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16)
+    )
+
+    class BinaryDataByteFormat(CustomEnum):
+        MSB_FIRST = "NORM"
+        LSB_FIRST = "SWAP"
+
+    class DataFileParameter(CustomEnum):
+        """Parameter format in a data file"""
+
+        LINEAR_AND_PHASE = "LINPH"
+        LOG_AND_PHASE = "LOGPH"
+        REAL_AND_IMAGINARY_NUMBERS = "REIM"
+
+    class BinaryDataFormat(CustomEnum):
+        """Binary Data file format."""
+
+        ASC = "ASC"
+        REAL = "REAL"
+        REAL32 = "REAL32"
+
+    class DataFileFrequencyUnits(CustomEnum):
+        """Binary Data file frequency units."""
+
+        HZ = "HZ"
+        KHZ = "KHZ"
+        MHZ = "MHZ"
+        GHZ = "GHZ"
```

Comment:
I'm not sure if I'm fully convinced of the use of these enums; could you elaborate a bit on the reason to use them instead of 'simply' using the strings? (especially since for most enums, the enum-value and the string-value are identical, e.g. `HZ = "HZ"`)

One of the things that I do dislike is exemplified in the test:
```
instr.datafile_frequency = AnritsuMS464xB.DataFileFrequencyUnits.KHZ
assert instr.datafile_frequency == str(AnritsuMS464xB.DataFileFrequencyUnits.KHZ)
```
So what I take from this is that the value/type that is returned from the property (a string) is not the same as what you set (an enum), which for me would be very confusing.

Another thing (which can/should be fixed by improving the docstrings) is that at present, as a user, it would be unclear to me what I should assign to a property if I want to write something.

Reply:
@ashokbruno , if you use `cast = AnritsuMS464xB.DataFileFrequencyUnits`, an enum will be returned, instead of the string.